x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/LocalStateCompositeXPackPlugin.java

```diff
@@ -476,4 +482,15 @@ public void close() throws IOException {
                 .collect(Collectors.toList());
     }
 
+    private Environment getUpdatedEnvironment(Environment existingEnvironment){
+        if (inFipsSunJsseJvm()) {
+            Settings additionalSettings = Settings.builder()
+                .put(existingEnvironment.settings())
+                .put(XPackSettings.DIAGNOSE_TRUST_EXCEPTIONS_SETTING.getKey(), false)
```

Comment:
Can't this be returnned from `additionalSettings()`?

Reply:
Apologies if I misunderstood but did you read through https://github.com/elastic/elasticsearch/pull/52211#issuecomment-587293981 ? I'll go and double check this again but this was supposed to show why additionalSettings doesn't solve this here

Reply:
Sorry I missed that explanation. It looks to me like a bug. We should be passing the final environment everywhere once it is created.

Reply:
I'll raise a PR to fix that then and change to use additionalSettings here

Reply:
I've opened https://github.com/elastic/elasticsearch/pull/52602 to cleanup the name to avoid this situation.